### PR TITLE
Fix oral histories citations

### LINKED ISSUE #1169: "Cite as" on Oral histories is not quite right
When we designed "Cite As", we had particular examplar/models, and we weren't doing OH yet. 

So eg https://digital.sciencehistory.org/works/gh93h041p

We show:

> Cite as: “Oral History Interview with William John Bailey,” June 3, 1986. https://digital.sciencehistory.org/works/gh93h041p."

The OPAC also has a manually entered suggested citation, and it differs/is better:

> Cite as: "William John Bailey, interviewed by James J. Bohing at the Univeristy of Maryland on 3 June 1986, (Philadelphia: Chemical Heritage Foundation, Oral History Transcript 0012.)"

It should not be hard to improve our citation display to be closer to that, if not exactly that. We have all of those data elements as machine-interprable data in the record, just need a special path to assemble them in the way we want for things marked Oral Histories (which they are as a 'genre' -- we already have genre-specific citation code). 

May also apply to RIS export improvements.

### PR DESCRIPTION
**Pull request by eddierubeiz**
closed #1169

### CODE REVIEW

File: app/models/chf/citable_attributes.rb [hunk 1]
```diff
@@ -475,5 +482,48 @@ def archive_location
...
+      end
+
+      def date
+        return nil if work.date_of_work.length == 0
+        begin
+          date_recorded = Date.strptime(work.date_of_work.first, '%Y-%m-%d')
+        rescue ArgumentError
+          return nil
+        end
+        date_recorded ? CiteProc::Date.new([date_recorded.year]) : nil
```
Comment (jrochkind) [inline app/models/chf/citable_attributes.rb]: i think we'd rather the whole date with month and day for these, not just the year. To more closely match suggested format from catalog. 

I'm not totally sure how you create a CiteProc::Date like that either, might have to look at CiteProc code in that gem.

**Comment by jrochkind** [comment]
Needs tests. No reason not to write em now, if you change the desired output, you'll find the existing tests a nice cushion, and you can even do TDD if you need changes.

File: app/models/chf/citable_attributes.rb [hunk 2]
```diff
@@ -35,9 +35,16 @@ def initialize(work,
       @collection = collection.nil? ? work.in_collection_presenters.first : collection
       @parent_work = parent_work.nil? ? work.parent_work_presenters.first : parent_work
       @edge_case_date_literals = !!edge_case_date_literals
-      @implementation = treat_as_local_photograph? ?
-        TreatAsLocalPhotograph.new(@work, collection: @collection, parent_work: @parent_work) :
-        StandardTreatment.new(@work, collection: @collection, parent_work: @parent_work)
+
+      if work.genre_string != nil && work.genre_string.include?('Oral histories')
```
Comment (jrochkind) [inline app/models/chf/citable_attributes.rb]: Extract this into a method to parallel the `treat_as_local_photograph?` line? Maybe `cite_as_oral_history?` ?

File: app/models/chf/citable_attributes.rb [hunk 3]
```diff
@@ -35,9 +35,16 @@ def initialize(work,
...
+        @implementation = TreatAsOralHistory.new(@work, collection: @collection, parent_work: @parent_work)
+      elsif treat_as_local_photograph?
+      #if treat_as_local_photograph?
```
Comment (jrochkind) [inline app/models/chf/citable_attributes.rb]: remove commented out code

**Comment by jrochkind** [comment]
ps nice work

File: app/models/chf/citable_attributes.rb [hunk 4]
```diff
@@ -475,5 +487,50 @@ def archive_location
       end
     end
 
+    class TreatAsOralHistory < StandardTreatment
+      def title
+        return work.title if work.interviewee.nil? || work.interviewer.nil?
+        time = work.place_of_interview.nil? ? "" : "at #{work.place_of_interview.first}"
+        place = date.nil? ? "" : "on #{date.iso8601}"
```
[the remark this answers is not in the document]
Comment (eddierubeiz) [reply app/models/chf/citable_attributes.rb]: Yup! Let me fix...

**Comment by jrochkind** [comment]
The examples in our catalog have "Oral History Transcript 0279" -- we do have that in our metadata, we can prob get it in the citation somehow. I might mess with it just to mess with it?

**Pull request merged**